corona-ide-ui/src/main/java/com/coronaide/ui/controller/MainController.java

```diff
@@ -79,10 +83,51 @@ private void prepareWorkspace() {
     }
 
     private void showProjectList() {
-        List<String> projectNamesList = projectService.getAll().stream().map(Project::getName)
-                .collect(Collectors.toList());
-        ObservableList<String> observableProjectNamesList = FXCollections.observableList(projectNamesList);
-        listViewProjects.setItems(observableProjectNamesList);
+        List<Project> projectsList = projectService.getAll().stream().collect(Collectors.toList());
+        ObservableList<Project> observableProjectsList = FXCollections.observableList(projectsList);
+
+        listViewProjects.setCellFactory(listView -> new ListCell<Project>() {
+            @Override
+            protected void updateItem(Project item, boolean empty) {
+                super.updateItem(item, empty);
+                if (!empty) {
+                    setText(item.getName());
+
+                    ContextMenu contextMenu = new ContextMenu();
+                    MenuItem menuDelete = new MenuItem();
+                    menuDelete.setText("Delete");
+                    menuDelete.setOnAction(event -> {
+                        AlertWithCheckbox alert = new AlertWithCheckbox(AlertType.CONFIRMATION,
+                                "Also delete files from disk (cannot be undone)", ButtonType.YES);
+                        alert.setTitle("Delete Project");
+                        alert.setContentText("Do you want to remove " + item.getName() + " from your workspace?");
+
+                        alert.showAndWait()
+                                .filter(r -> r == ButtonType.YES)
+                                .ifPresent(r -> {
```

Comment:
This is a gaint thing to be putting inline in a lambda like this - can we instead make it a private method or something and reference it?